lib/classes/Variables.js

```diff
@@ -308,7 +308,13 @@ class Variables {
     return BbPromise.all(populations).then((results) =>
       results.forEach((result) => {
         if (result.value !== result.populated) {
-          _.set(target, result.path, result.populated);
+          // We could not use direct `_.set` as due to security measures, it did not allow setting values
+          // if `prototype` was a part of the path and that caused an infinite loop. In our use-case, there are no
+          // security vulnerabilities.
+          const pathArray = Array.isArray(result.path) ? result.path : result.path.split('.');
+          let obj = target;
+          for (const property of pathArray.slice(0, -1)) obj = obj[property];
+          obj[pathArray.slice(-1)] = result.populated;
```

Comment:
In last line we pass array as property name, and it works by a chance of array stringification.

I would probably literally reference last property via e.g. `_.last(pathArray)`

Reply:
good call, changed